sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -269,7 +277,7 @@ object FileSourceStrategy extends Strategy with PredicateHelper with Logging {
             case FileFormat.FILE_PATH | FileFormat.FILE_NAME | FileFormat.FILE_SIZE |
                  FileFormat.FILE_MODIFICATION_TIME =>
               col
-            case FileFormat.ROW_INDEX =>
+            case FileFormat.ROW_INDEX_TEMPORARY_COLUMN_NAME =>
```

Comment:
is it a bug fix?

Reply:
This is an effect of the different iteration / building of different lists that I apply above.
Before, metadataColumns was just a flattened list of all columns.
Now the flattening also includes all transformations. I introduced this to have one place in the code that handles the different types only.